### PR TITLE
[wix-style-react] - Fix Sidebar gradient not shown on inner menu drilldown

### PR DESCRIPTION
**Pull request by miby-wix**
### 🔦 Summary

Refactored sidebar - added a wrapper to the content children to add gradient on both main menu and inner menus.
Added test as well.

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [V] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6561

File: src/Sidebar/SidebarContentWrapper.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { st, classes } from './Sidebar.st.css';
+import { dataHooks, sidebarSkins } from './constants';
+
+/** A sidebar content children wrapper to add gradient  */
+export class SidebarContentWrapper extends Component {
+  static propTypes = {
+    skin: PropTypes.oneOf(['dark', 'light']),
```
Comment (gendeld) [inline src/Sidebar/SidebarContentWrapper.js:9]: You're missing some props here.
Also, I don't see you passing the skin prop from sidebar

File: src/Sidebar/SidebarContentWrapper.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  _shouldAddGradient() {
+    const { scrollHeight, clientHeight } = this.childrenContainerRef.current;
+    this.setState({ isScrollbarDisplayed: scrollHeight > clientHeight });
+  }
+
+  render() {
+    const { skin, containerClasses, children, containerDataHook } = this.props;
+    const css = { ...classes, ...this.props.classNames };
```
Comment (gendeld) [inline src/Sidebar/SidebarContentWrapper.js:74]: I'm not sure what you are trying to do here

File: src/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -299,32 +246,18 @@ class Sidebar extends Component {
...
-              {this.state.isScrollbarDisplayed && (
-                <div
-                  className={gradientClasses}
-                  data-hook={dataHooks.scrollBarGradient}
-                />
-              )}
-            </div>
+            <SidebarContentWrapper
+              containerClasses={sliderClasses}
+              containerDataHook={''}
```
Comment (gendeld) [inline src/Sidebar/Sidebar.js:251]: You can probably just not pass it

File: src/Sidebar/SidebarContentWrapper.js (hunk 2 above)
Comment (miby-wix) [reply src/Sidebar/SidebarContentWrapper.js:74]: I took it as is from sidebar class on refactoring. It's just to keep convention on how to access css classes.

File: src/Sidebar/SidebarContentWrapper.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      skin,
+    });
+
+    return (
+      <div
+        className={css.containerClasses}
+        ref={this.childrenContainerRef}
+        data-hook={containerDataHook}
+      >
+        <div className={st(css.childrenContent)} ref={this.childrenContentRef}>
```
Comment (gendeld) [inline src/Sidebar/SidebarContentWrapper.js:89]:
```suggestion
        <div className={css.childrenContent} ref={this.childrenContentRef}>
```

**Pull request closed without merging**